guide/kong_integration.md

```diff
@@ -0,0 +1,96 @@
+# Using Kong as an API gateway for OpenFaaS
+
+[Kong](https://getkong.org) is an API gateway that provides features such as security, logging, and rate limiting. By putting this in front of OpenFaaS you can quickly get access to these things and a lot more via [the many other plugins written](https://getkong.org/plugins/) for it.
+
+Below is a demo of how you could use Kong as an authentication layer for OpenFaaS. You should be able to paste this all (from its Markdown source) into [Play With Docker](http://labs.play-with-docker.com/) to see it in action.
+
+## Setup OpenFaaS
+
+    docker swarm init --advertise-addr $(hostname -i)
```

Comment:
General established documentation format is to use triple block-quotes ``` and to prefix input with `$` and output without any prefix.

The services look good :+1: I think there is also a chance we could use a Docker Stack here.

Reply:
I can change to block quotes and input and outputs. But my intent as stated above is that you can cut and paste this whole from markdown source into play-with-docker or your local terminal to test it.

Thats how I built and and thats how I test it. Since headings are prefixed with hash marks they are interpreted as comments. If I put in block quotes each section will have to be separately pasted by the user. If we start each command with $, then even more cutting and pasting would be needed.

Actually what would be really nice would be implementing a PWD lab with clickable text sections so the user doesnt even have to cut and paste - but then again you can see I'm a little over excited about PWD :-)

I guess I favor keeping this doc easily testable, but I if someone feels this diverges too much from an established documentation format, I'm OK with changing it.

Reply:
I know blocks can't be copied and pasted but it's the established style and slows the tutorial down too. This is in-line with the Docker and Arch Linux documentation.

You can submit a PWD lab for this, I wrote one for multi-stage builds that you can use as a template.

Reply:
Concerning the docker stack... are you talking about modifying the faas/docker-compose.yml file to include kong? Or maybe creating something like docker-coompose.secure.yml that will do this?

Reply:
I can imagine this being a core use-case for people. In particular Kubernetes also uses a "stacks" concept through YAML files. We should test this out against FaaS-netes and provide a "YAML" configuration similar to: https://github.com/alexellis/faas-netes/blob/master/monitoring.yml

Reply:
Hm. Perhaps a separate tight for that one.